src/Adapter/Session/Repository/EmployeeSessionRepository.php

```diff
@@ -0,0 +1,98 @@
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Session\Repository;
+
+use EmployeeSession;
+use PrestaShop\PrestaShop\Adapter\AbstractObjectModelRepository;
+use PrestaShop\PrestaShop\Core\Domain\Security\Exception\CannotBulkDeleteEmployeeSessionException;
+use PrestaShop\PrestaShop\Core\Domain\Security\Exception\CannotDeleteEmployeeSessionException;
+use PrestaShop\PrestaShop\Core\Domain\Security\Exception\SessionNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Security\ValueObject\EmployeeSessionId;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+
+/**
+ * Methods to access data storage for Employee session
+ */
+class EmployeeSessionRepository extends AbstractObjectModelRepository
+{
+    /**
+     * @param EmployeeSessionId $sessionId
+     *
+     * @return EmployeeSession
+     *
+     * @throws CoreException
+     */
+    public function get(EmployeeSessionId $sessionId): EmployeeSession
+    {
+        /** @var EmployeeSession $session */
+        $session = $this->getObjectModel(
+            $sessionId->getValue(),
+            EmployeeSession::class,
+            SessionNotFoundException::class
+        );
+
+        return $session;
+    }
+
+    /**
+     * @param EmployeeSessionId $employeeSessionId
+     *
+     * @throws CoreException
+     */
+    public function delete(EmployeeSessionId $employeeSessionId): void
+    {
+        $this->deleteObjectModel($this->get($employeeSessionId), CannotDeleteEmployeeSessionException::class);
+    }
+
+    /**
+     * @param array $employeeSessionIds
+     *
+     * @throws CannotBulkDeleteEmployeeSessionException
+     */
+    public function bulkDelete(array $employeeSessionIds): void
+    {
+        $failedIds = [];
+        foreach ($employeeSessionIds as $employeeSessionId) {
+            try {
+                $this->delete($employeeSessionId);
+            } catch (CannotDeleteEmployeeSessionException $e) {
+                $failedIds[] = $employeeSessionId->getValue();
+            }
+        }
+
+        if (empty($failedIds)) {
+            return;
+        }
+
+        throw new CannotBulkDeleteEmployeeSessionException(
```

Comment:
There's something wrong with the way you create this exception You us an array as first parameter, but a string message is expected So it will likely raise an error

The way it is displayed in the Controller (in `getErrorMessages`) is not correct either, you insert the message from the exception which means it will always be in english

You can inspire yourself from `CannotBulkDeleteTaxRulesGroupException` which uses the array of failedIds as first parameter, and for the error message you can see how it's done here:

https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Controller/Admin/Improve/International/TaxRulesGroupController.php#L319

Reply:
I believe all your bulk exceptions have the same issue, so they all need this fix

Reply:
I reproduce the exact same code as @zuk3975 showed me :sweat_smile: 
https://github.com/PrestaShop/PrestaShop/blob/develop/src/Adapter/Product/Repository/ProductRepository.php#L336
https://github.com/PrestaShop/PrestaShop/blob/develop/src/Adapter/Product/CommandHandler/BulkDeleteProductHandler.php#L38

Reply:
My bad, I didn't follow the exception constructor in the ProductRepositoryu :sweat_smile:

Reply:
Done